packages/@ourworldindata/grapher/src/core/LegacyToOwidTable.ts

```diff
@@ -359,7 +327,36 @@ export const legacyToOwidTableAndDimensions = (
         }
     }
 
-    return { dimensions: newDimensions, table: joinedVariablesTable }
+    // Append the entity color column if we have selected entity colors
+    if (!isEmpty(selectedEntityColors)) {
+        const entityColorColumnSlug = OwidTableSlugs.entityColor
+
+        const valueFn = (
+            entityId: EntityId | undefined
+        ): string | ErrorValue => {
+            if (!entityId) return ErrorValueTypes.UndefinedButShouldBeString
+            const entityName =
+                joinedVariablesTable.entityIdToNameMap.get(entityId)
+            return entityName && selectedEntityColors
+                ? (selectedEntityColors[entityName] ??
+                      ErrorValueTypes.UndefinedButShouldBeString)
+                : ErrorValueTypes.UndefinedButShouldBeString
+        }
+
+        const values = joinedVariablesTable.rows.map((row) =>
+            valueFn(row.entityId)
+        )
```

Comment:
perf: computing `.rows` is nontrivial (since our column store is column-based), and we should avoid it if we can.

you can instead use something like
```ts
joinedVariablesTable.entityNameColumn.valuesIncludingErrorValues.map()
```

Reply:
Good point, thanks!